src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -91,105 +108,137 @@ static EnAobStruct D_809C384C[] = {
     { { -4827.0f, 168.0f, 1328.0f }, 115, 4 }, { { -4130.0f, 150.0f, 1367.0f }, 112, 0 },
 };
 
-void func_809C10B0(EnAob01* this, s32 arg1) {
+void EnAob01_Blink(EnAob01* this, s32 indexToRestartBlink) {
     if (DECR(this->blinkTimer) == 0) {
         this->eyeIndex++;
-        if (this->eyeIndex >= arg1) {
-            this->eyeIndex = 0;
+        if (this->eyeIndex >= indexToRestartBlink) {
+            this->eyeIndex = EN_AOB01_EYE_OPEN;
             this->blinkTimer = Rand_S16Offset(30, 30);
         }
     }
 }
 
-void func_809C1124(void) {
+/**
+ * Called every frame during the race in order to make in-game time pass.
+ */
+void EnAob01_AdvanceTime(void) {
     gSaveContext.save.time = ((void)0, gSaveContext.save.time) + (u16)R_TIME_SPEED;
     gSaveContext.save.time = ((void)0, gSaveContext.save.time) + (u16)((void)0, gSaveContext.save.timeSpeedOffset);
 }
 
-void func_809C1158(EnAob01* this, PlayState* play) {
-    s32 temp_s0 = ENAOB01_GET_7E00_2(&this->actor);
+/**
+ * Initializes all the paths that the dogs follow.
+ */
+void EnAob01_InitializeDogPaths(EnAob01* this, PlayState* play) {
+    s32 pathIndex = ENAOB01_GET_STARTING_DOG_PATH_INDEX(&this->actor);
     s16 i = 0;
 
-    if (temp_s0 != 0x3F) {
+    if (pathIndex != 0x3F) {
         do {
-            this->unk_1D8[i] = SubS_GetPathByIndex(play, temp_s0, 0x3F);
-            temp_s0 = this->unk_1D8[i]->unk1;
+            this->dogPaths[i] = SubS_GetPathByIndex(play, pathIndex, 0x3F);
+            pathIndex = this->dogPaths[i]->unk1;
             i++;
-        } while (temp_s0 != 0xFF);
+        } while (pathIndex != 0xFF);
     }
```

Comment:
Which path macros? The ones in PR here? https://github.com/zeldaret/mm/pull/1205

Reply:
As for the other part of your message, a plain-while adds this extra `beq` instruction:
![image](https://user-images.githubusercontent.com/12245827/234114474-985fa404-754d-4436-a2c1-f685b2b3651b.png)

Reply:
It is probably because of the different comparisons being made. The first path index is coming from her individual paras, so is only 0x3F (5 bit -1) for the NONE_PATH_INDEX. On the other hand the additional path indices from the scene have a NONE_PATH_INDEX of 0xFF (8 bit -1). Technically the fact they use `0x3F` in the function call is wrong after the first iteration, since a none would be 0xFF at that point.